test-data/unit/check-isinstance.test

```diff
@@ -1788,6 +1788,20 @@ else:
 [builtins fixtures/list.pyi]
 [out]
 
+[case testNarrowTypeAfterInListOfOptional]
+# flags: --strict-optional
+from typing import List, Optional
+
+x: List[Optional[int]]
```

Comment:
Here's another special case: What if the container is a nested one, say `List[List[x]]`, where `x` might be `Any`? How do we deal with various item types, such as `List[int]` and `List[Any]`?

Reply:
a) If the container is ``List[Any]``, then we do nothing (there is a test for this).
b) If the container is ``List[List[Any]]`` we narrow the type (provided there is an overlap, this is consistent with how ``==`` currently treated), for example:
```python
x: Optional[int]
lst: Optional[List[int]]
nested: List[List[Any]]
if lst in nested:
    reveal_type(lst) # List[int]
if x in nested:
    reveal_type(x) # Optional[int]
```
There is already a test for non-overlapping items ``int`` vs ``str``. I will add one more test for other (nested) types?

Reply:
OK, added one more test.